File: clients/rust/model/src/weapon.rs
use crate::*;
#[derive(Clone, Debug, trans::Trans)]
pub struct Weapon {
    pub typ: WeaponType,
    pub params: WeaponParams,
    pub magazine: i32,
    pub was_shooting: bool,
    pub spread: f64,
    pub fire_timer: Option<f64>,
    pub last_angle: Option<f64>,
    pub last_fire_tick: Option<i32>,
}

File: clients/haskell/src/RAIC/Model/Weapon.hs
module RAIC.Model.Weapon where

import           GHC.Generics            (Generic)
import           RAIC.Model.WeaponParams (WeaponParams)
import           RAIC.Model.WeaponType   (WeaponType)
import           RAIC.Utils.Trans        (Trans)

data Weapon = Weapon {
  weapon_type    :: WeaponType,
  params         :: WeaponParams,
  magazine       :: Int,
  was_shooting   :: Bool,
  spread         :: Double,
  file_timer     :: Maybe Double,
  last_angle     :: Maybe Double,
  last_fire_tick :: Maybe Int
} deriving (Generic, Show)

instance Trans Weapon
